src/views/operations/useGetOperations.tsx

```diff
@@ -18,103 +19,102 @@ import {
+      limit?: number;
+      sort?: "asc" | "desc";
+    }
+  | undefined;
+
+/**
+ * Hook to fetch operations for given addresses.
+ *
+ * It fetches the latest operations on the first load and allows
+ * to load more (older) operations as needed.
+ *
+ * Every {@link REFRESH_INTERVAL} it fetches the latest operations
+ * and prepends it to the already fetched operations list.
+ *
+ * If the refresh button is clicked then it will trigger this behaviour immediately.
+ *
+ * @param addresses - list of addresses to fetch operations related to
+ * @returns
+ *   operations - operations ordered by id in descending order
+ *   isFirstLoad - true if the first load is in progress
+ *   isLoading - true if the next page is being fetched
+ *   hasMore - true if there are more operations to fetch
+ *   loadMore - function to load more operations (older ones)
+ */
+export const useGetOperations = (addresses: RawPkh[]) => {
   const network = useSelectedNetwork();
-  const [operations, setOperations] = useState<TzktCombinedOperation[]>([]);
-  const [hasMore, setHasMore] = useState(true);
-  const [isFirstLoad, setIsFirstLoad] = useState(true);
-  const { isLoading, handleAsyncAction } = useAsyncActionHandler();
-
-  const [addresses, setAddresses] = useState<RawPkh[]>(initialAddresses);
   const dispatch = useAppDispatch();
+  const refetchTrigger = useRefetchTrigger();
 
-  const [updatesTrigger, setUpdatesTrigger] = useState(0);
+  const {
+    isFetching,
+    data: operations,
+    hasNextPage,
+    isLoading,
+    fetchNextPage,
+    fetchPreviousPage,
+  } = useInfiniteQuery({
+    queryKey: ["operations", addresses, dispatch, network],
+    initialPageParam: undefined as QueryParams,
+    queryFn: async ({ pageParam }) =>
+      fetchOperationsAndUpdateTokensInfo(dispatch, network, addresses, pageParam),
+    getNextPageParam: lastPage => {
+      if (lastPage.length === 0) {
+        return undefined;
+      }
+
+      const lastId = lastPage[lastPage.length - 1].id;
+      return { lastId };
+    },
+
+    getPreviousPageParam: (_, pages) => {
+      const lastId = maxBy(pages.flat(), "id")?.id;
+
+      // if lastId is not defined it means that there are no operations yet for the accounts
+      // but we still need to keep updating the operations list to get ones once they appear
+      return lastId ? { lastId, sort: "asc" as const } : undefined;
+    },
+
+    select: ({ pages }) => {
+      const firstPage = pages[0];
+      // no need to shuffle the array if there are no operations
+      // it happens when there are no new operations since last fetch
+      // it will add an empty page to the list of pages
+      if (firstPage.length === 0) {
+        return pages.flat();
+      }
+
+      // new operations are always sorted ascending by id
+      // but they are displayed in descending order
+      firstPage.sort((a, b) => b.id - a.id);
+
+      return filterDuplicatedTokenTransfers([...firstPage, ...pages.slice(1).flat()]);
```

Comment:
Why we using slice(1) here?

Reply:
because the first page was changed only, everything starting with the index 1 has to be copied over